src/main.js

```diff
@@ -608,7 +609,12 @@ define([
     self.once('player.canplay', playerCanPlayHandler);
     self.once('player.error', playerErrorHandler);
 
-    self.options.mediaElement.setAttribute('src', options.mediaUrl);
+    if (options.mediaUrl) {
```

Comment:
I think this should check for `this.options.mediaElement`. It amounts to the same thing, but seems clearer to me (If there's a media element, change it's source and wait for a `canplay` event, if there's a custom player, create the waveform directly without waiting for an event).